### PR TITLE
Fixturize tests/frame/test_asof.py

### PR DESCRIPTION
**Pull request by h-vetinari**
Picking up #22471 again. Added one new fixture.

### CODE REVIEW

File: pandas/tests/frame/conftest.py [hunk 1]
```diff
@@ -67,6 +67,18 @@ def datetime_frame():
     return DataFrame(tm.getTimeSeriesData())
 
 
+@pytest.fixture
+def date_range_frame():
+    """
+    Fixture for DataFrame of ints with date_range index
```
Comment (jreback) [inline pandas/tests/frame/conftest.py]: not in love with this as a generalized fixture, if you put in test_asof would be ok

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/frame/conftest.py (hunk 1 above)
Comment (h-vetinari) [reply pandas/tests/frame/conftest.py]: So far, the requirement was always that this should be on `tests/frame/conftest` to be reusable for other test modules. But I don't care either way - can put it into `test_asof.py` as well.

**Comment by h-vetinari** [comment]
@jreback 
Moved the fixture to the `test_asof` module, and it's green.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks

**Pull request merged**